src/grid/column.directive.ts

```diff
@@ -0,0 +1,164 @@
+import {
+	Directive,
+	HostBinding,
+	Input,
+	OnChanges,
+	OnDestroy,
+	OnInit,
+	Optional
+} from "@angular/core";
+import { Subscription } from "rxjs";
+import { GridService } from "./grid.service";
+
+@Directive({
+	selector: "[cdsCol], [ibmCol]"
+})
+export class ColumnDirective implements OnInit, OnChanges, OnDestroy {
+	@HostBinding("class")
+	get columnClasses(): string {
+		return this._columnClasses.join(" ");
+	}
+	set columnClasses(classes: string) {
+		this._columnClasses = classes.split(" ");
+	}
+
+	@Input() class = "";
+
+	/**
+	 * Defines columns width for specified breakpoint
+	 * Accepts the following formats:
+	 * - {[breakpoint]: number}
+	 * - {[breakpoint]: "auto"} - css only
+	 * - {[breakpoint]: {[start|end]: number}} - css only
+	 *
+	 * Example:
+	 * <div cdsCol [columnNumbers]={md: 3, lg: 4}></div>
+	 */
+	@Input() columnNumbers = {};
+
+	/**
+	 * Defines columns offset, which increases the left margin of the column.
+	 * This field will only work with flexbox grid.
+	 *
+	 * Accepts the following formats:
+	 * - {[breakpoint]: number}
+	 *
+	 * Example:
+	 * <div cdsCol [offsets]={md: 3, lg: 4}></div>
+	 */
+	@Input() offsets = {};
+
+	/**
+	 * Set to `true` to use css grid column hang class
+	 * This will only work when `isCss` property is set to true
+	 *
+	 * Useful when trying to align content across css grid/subgrid
+	 */
+	@Input() columnHang = false;
+
+	protected _columnClasses: string[] = [];
+
+	private isCssGrid = false;
+	private subscription = new Subscription();
+
+	constructor(@Optional() private gridService: GridService) {}
+
+	ngOnInit() {
+		if (this.gridService) {
+			this.gridService.gridObservable.subscribe((isCssGrid: boolean) => {
```

Comment:
Isn't this missing `this.subscription = ` to unsubscribe on destroy?

Reply:
Thanks for pointing this out @eduardmarcinco. A fix has been merged to address this.